src/jenkins.py

```diff
@@ -0,0 +1,87 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Functions to operate Jenkins."""
+
+from datetime import datetime, timedelta
+from pathlib import Path
+from time import sleep
+from typing import cast
+
+import requests
+
+from types_ import Credentials, JenkinsEnvironmentMap
+
+JENKINS_WEB_URL = "http://localhost:8080"
+JENKINS_HOME = Path("/var/lib/jenkins")
+
+
+def _is_jenkins_ready() -> bool:
+    """Check if Jenkins webserver is ready.
+
+    Returns:
+        True if Jenkins server is online. False otherwise.
+    """
+    try:
+        return requests.get(f"{JENKINS_WEB_URL}/login", timeout=10).ok
+    except requests.exceptions.ConnectionError:
```

Comment:
Is this the only kind of error that can occur here?

Reply:
Updated to include TimeoutError, thank you!